src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -702,10 +702,13 @@ protected DynamoModel(IStartConfiguration config)
                     // Do nothing for now
                 }
             }
+            //align disable analytics with nonetwork mode, this is done in other entry points(CLI,Sandbox etc) - but
+            //not all integrators will use those entry points, some may just create a DynamoModel directly.
+            Analytics.DisableAnalytics = NoNetworkMode;
```

Comment:
Do we also need to check for cmd line args `DisableAnalytics` flag here?

Reply:
no, command line args may not exist, someone could just create a DynamoModel directly without using any CLI.

Reply:
I meant if we should add a case for CLI here.

Reply:
On second thoughts, I don't think these flags should be conflated with one another. `DisableAnalytics` should be able to be set to `true` even if `NoNetworkMode` is set to `false`. 

Adding the additional check for `!NoNetworkMode` below should suffice.